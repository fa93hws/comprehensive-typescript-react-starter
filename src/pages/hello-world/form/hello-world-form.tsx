import * as React from 'react';
import styles from './hello-world-form.css';

interface HelloWorldFormProps {
  PasswordField: React.ComponentType;
  PasswordConfirmationField: React.ComponentType;
}

export const HelloWorldForm = (props: HelloWorldFormProps) => (
  <form className={styles.form}>
    <props.PasswordField/>
    <props.PasswordConfirmationField/>
  </form>
);
