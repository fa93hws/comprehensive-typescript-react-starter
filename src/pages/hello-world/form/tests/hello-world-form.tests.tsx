import * as React from 'react';
import { HelloWorldForm } from '../hello-world-form';

describe('HelloWorldForm', () => {
  it('should render correctly', () => {
    const PasswordField = () => <div>Password</div>;
    const PasswordConfirmation = () => <div>Password Confirmation</div>;
    expect(
      <HelloWorldForm
        PasswordField={PasswordField}
        PasswordConfirmationField={PasswordConfirmation}
      />,
    ).toMatchRenderedSnapshot();
  });
});
