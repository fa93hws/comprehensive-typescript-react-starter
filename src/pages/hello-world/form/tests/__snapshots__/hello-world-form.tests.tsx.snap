// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldForm should render correctly 1`] = `
<form>
  <div>
    Password
  </div>
  <div>
    Password Confirmation
  </div>
</form>
`;
